### ISSUE TITLE
IPFS will sometimes crash upon initialisation if swarm addresses are provided.

### ISSUE THREAD

**Issue by HexaField** [ipfs/js-ipfs#3182]
- **Version**:

- **Platform**:

**Comment by achingbrain** [comment]
Can you post your package.json? You may have some libp2p modules that need to be updated.

**Comment by HexaField** [comment]
The only ipfs related packages I have are
    "ipfs": "0.48.0",
    "ipfs-pubsub-room": "^2.0.1",
    "libp2p-pubsub": "0.5.3",

**Comment by achingbrain** [comment]
@vasco-santos the crash is caused by a non-null value being returned from the libp2p connection manager - I'm guessing [this line](https://github.com/ipfs/js-ipfs-bitswap/blob/master/src/network.js#L56) needs to change from:

```javascript
conn && this._onPeerConnect(conn)
```

to

```javascript
conn && conn.remotePeer && this._onPeerConnect(conn.remotePeer)
```

**Comment by vasco-santos** [comment]
Yes, `onPeerConnect` should receive the PeerId. The validation of the `conn.remotePeer` should not be needed as we only add a connection to the connMgr when it was upgraded and will have the remotePeer always.
If you are experiencing a scenario that does not include it, we might have a bug somewhere. Let me know

**Comment by achingbrain** [comment]
This should be fixed in `v0.48.1` - please update and reopen if you see this crash again.
